File: C++/1046. Last Stone Weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq;
        for(int i=0;i<stones.size();i++){
            pq.push(stones[i]);
        }
        while(pq.size()!=1){
            int i = pq.top();
            pq.pop();
            int j = pq.top();
            pq.pop();
            pq.push(i - j);
        }
        return pq.top();
    }
};

File: C#/1046. Last Stone Weight.cs
public class Solution {
    public int LastStoneWeight(int[] stones) {
        var pq = new PriorityQueue<int, int>();
        for(int i=0;i<stones.Length;i++){
            pq.Enqueue(stones[i], int.MaxValue - stones[i]);
        }
        while(pq.Count!=1){
            int i = pq.Dequeue();
            int j = pq.Dequeue();
            pq.Enqueue(i - j, int.MaxValue - i + j);
        }
        return pq.Dequeue();
    }
}
